dataprofiler/tests/labelers/test_base_model.py

```diff
@@ -96,25 +96,98 @@ def test_set_parameters(self):
         with self.assertRaisesRegex(ValueError, 'test'):
             base_model.BaseModel.set_params(mock_model, **params)
 
-    def test_set_label_mapping_parameters(self):
+    def test_add_labels(self):
 
         # setup mock
         mock_model = mock.Mock(spec=base_model.BaseModel)
-
-        # base case
+        mock_model._convert_labels_to_label_mapping.side_effect = \
+            base_model.BaseModel._convert_labels_to_label_mapping
+        mock_model._label_mapping = {}
+
+        # assert bad label inputs
+        with self.assertRaisesRegex(TypeError, '`label` must be a str.'):
+            base_model.BaseModel.add_label(mock_model, label=None)
+
+        with self.assertRaisesRegex(TypeError, '`label` must be a str.'):
+            base_model.BaseModel.add_label(mock_model, label=1)
+
+        # assert existing label
+        label = 'NEW_LABEL'
+        mock_model._label_mapping = {'NEW_LABEL': 1}
+        with self.assertWarnsRegex(UserWarning,
```

Comment:
more specific warning than UserWarning?

Reply:
I do not think this should be separate from user as it is directly created by the user's add here.